pkg/native/serializers/serializer_cdx.go

```diff
@@ -224,14 +240,42 @@ func (s *SerializerCDX) nodeToComponent(n *sbom.Node) *cdx.Component {
 	}
 	c := &cdx.Component{
 		BOMRef:      n.Id,
-		Type:        cdx.ComponentType(strings.ToLower(n.PrimaryPurpose)), // Fix to make it valid
 		Name:        n.Name,
 		Version:     n.Version,
 		Description: n.Description,
 	}
 
-	if n.Type == sbom.Node_FILE {
-		c.Type = "file"
+	switch strings.ToLower(n.PrimaryPurpose) {
+	case "application":
+		c.Type = cdx.ComponentTypeApplication
+	case "container":
+		c.Type = cdx.ComponentTypeContainer
+	case "data":
+		c.Type = cdx.ComponentTypeData
+	case "device":
+		c.Type = cdx.ComponentTypeDevice
+	case "device-driver":
+		c.Type = cdx.ComponentTypeDeviceDriver
+	case "file":
+		c.Type = cdx.ComponentTypeFile
+	case "firmware":
+		c.Type = cdx.ComponentTypeFirmware
+	case "framework":
+		c.Type = cdx.ComponentTypeFramework
+	case "library":
+		c.Type = cdx.ComponentTypeLibrary
+	case "machine-learning-model":
+		c.Type = cdx.ComponentTypeMachineLearningModel
+	case "operating-system":
+		c.Type = cdx.ComponentTypeOS
+	case "platform":
+		c.Type = cdx.ComponentTypePlatform
+	case "":
+		// no node PrimaryPurpose set
+	default:
+		// Data loss here.
+		// TODO how do we handle when data loss occurs?
```

Comment:
We've been collecting instances of data loss with a comment like this:

```suggestion
		// TODO(degradation): Non-matching primary purpose to component type mapping
```